src/product.ts

```diff
@@ -13,9 +14,8 @@ import { purry } from "./purry";
  * @dataFirst
  * @category Number
  */
-export function product<
-  T extends ReadonlyArray<bigint> | ReadonlyArray<number>,
->(data: T): T[number];
+export function product(data: ReadonlyArray<number>): number;
+export function product(data: Readonly<NonEmptyArray<bigint>>): bigint;
```

Comment:
**question:** why is one `ReadonlyArray` and one `Readonly<NonEmptyArray<...>>`?